eva_cttv_pipeline/clinvar_to_evidence_strings.py

```diff
@@ -1,291 +1,284 @@
+            "n_multiple_allele_origin": 0, "n_germline_somatic": 0, "n_records_no_recognised_allele_origin": 0,
+            "no_variant_to_ensg_mapping": 0, "n_more_than_one_efo_term": 0, "n_same_ref_alt": 0,
+            "n_missed_strings_unmapped_traits": 0, "n_nsvs": 0, "n_valid_rs_and_nsv": 0, "n_nsv_skipped_clin_sig": 0,
+            "n_nsv_skipped_wrong_ref_alt": 0, "record_counter": 0, "n_total_clinvar_records": 0}
+
+
+def launch_pipeline(dir_out, allowed_clinical_significance, ignore_terms_file, adapt_terms_file, efo_mapping_file,
+                    snp_2_gene_file, variant_summary_file):
+
+    allowed_clinical_significance = allowed_clinical_significance.split(',') if allowed_clinical_significance else \
+        get_default_allowed_clincal_significance()
+
+    mappings = get_mappings(efo_mapping_file, ignore_terms_file, adapt_terms_file, snp_2_gene_file,
+                            variant_summary_file)
+
+    report = clinvar_to_evidence_strings(allowed_clinical_significance, mappings)
+
+    output(report, dir_out)
+
+
+def output(report, dir_out):
+    report.write_output(dir_out)
+    print(report)
+
+
+def clinvar_to_evidence_strings(allowed_clinical_significance, mappings):
+
+    report = Report(mappings.unavailable_efo_dict)
+
+    for cellbase_record in cellbase_records.get_records():
+
+        record = create_record(cellbase_record, mappings)
+        report.counters["record_counter"] += 1
+        report.counters["n_nsvs"] += (record.clinvarRecord.get_nsv(mappings.rcv_to_nsv) is not None)
+        append_nsv(report.nsv_list, record.clinvarRecord, mappings.rcv_to_nsv)
+
+        if skip_record(record, allowed_clinical_significance, mappings.rcv_to_nsv, report.counters):
+            continue
+
+        report.counters["n_multiple_allele_origin"] += (len(record.clinvarRecord.allele_origins) > 1)
+        report.counters["n_germline_somatic"] += (('germline' in record.clinvarRecord.allele_origins) and (
+        'somatic' in record.clinvarRecord.allele_origins))
+        report.counters["n_records_no_recognised_allele_origin"] += (
+            ('germline' not in record.clinvarRecord.allele_origins) and
+            ('somatic' not in record.clinvarRecord.allele_origins))
+
+        traits = create_traits(record.clinvarRecord.traits, mappings.trait_2_efo, report)
+
+        for ensembl_gene_id, trait, allele_origin in itertools.product(record.con_type.ensembl_gene_ids, traits,
+                                                                       record.clinvarRecord.allele_origins):
 
-                else:
-                    n_same_ref_alt += 1
-                    if clinvarRecord.get_nsv(rcv_to_nsv) is not None:
-                        n_nsv_skipped_wrong_ref_alt += 1
+            if allele_origin not in ('germline', 'somatic'):
+                report.n_unrecognised_allele_origin[allele_origin] += 1
+                continue
```

Comment:
Use `continue` or `else`, but having both is redundant. Just as a reminder, the suggestion was to use `continue` and remove the `else`, reducing indentation as a result.